File: app/src/main/java/org/schabi/newpipe/database/playlist/PlaylistMetadataEntry.java
package org.schabi.newpipe.database.playlist;

import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;

import static org.schabi.newpipe.database.playlist.model.PlaylistEntity.PLAYLIST_DISPLAY_INDEX;
import static org.schabi.newpipe.database.playlist.model.PlaylistEntity.PLAYLIST_ID;
import static org.schabi.newpipe.database.playlist.model.PlaylistEntity.PLAYLIST_NAME;
import static org.schabi.newpipe.database.playlist.model.PlaylistEntity.PLAYLIST_THUMBNAIL_URL;

public class PlaylistMetadataEntry implements PlaylistLocalItem {
    public static final String PLAYLIST_STREAM_COUNT = "streamCount";

    @ColumnInfo(name = PLAYLIST_ID)
    public final long uid;
    @ColumnInfo(name = PLAYLIST_NAME)
    public final String name;
    @ColumnInfo(name = PLAYLIST_THUMBNAIL_URL)
    public final String thumbnailUrl;
    @ColumnInfo(name = PLAYLIST_DISPLAY_INDEX)
    public long displayIndex;
    @ColumnInfo(name = PLAYLIST_STREAM_COUNT)
    public final long streamCount;

    public PlaylistMetadataEntry(final long uid, final String name, final String thumbnailUrl,
                                 final long displayIndex, final long streamCount) {
        this.uid = uid;
        this.name = name;
        this.thumbnailUrl = thumbnailUrl;
        this.displayIndex = displayIndex;
        this.streamCount = streamCount;
    }

    @Override
    public LocalItemType getLocalItemType() {
        return LocalItemType.PLAYLIST_LOCAL_ITEM;
    }

    @Override
    public String getOrderingName() {
        return name;
    }

    @Override
    public long getDisplayIndex() {
        return displayIndex;
    }
    @Nullable
    @Override
    public String getThumbnailUrl() {
        return thumbnailUrl;
    }
}

File: app/src/main/java/org/schabi/newpipe/database/playlist/PlaylistMetadataEntry.kt
/*
 * SPDX-FileCopyrightText: 2018-2025 NewPipe contributors <https://newpipe.net>
 * SPDX-FileCopyrightText: 2025 NewPipe e.V. <https://newpipe-ev.de>
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package org.schabi.newpipe.database.playlist

import androidx.room.ColumnInfo
import org.schabi.newpipe.database.LocalItem.LocalItemType
import org.schabi.newpipe.database.playlist.model.PlaylistEntity

open class PlaylistMetadataEntry(
    @ColumnInfo(name = PlaylistEntity.PLAYLIST_ID)
    override val uid: Long,

    @ColumnInfo(name = PlaylistEntity.PLAYLIST_NAME)
    override val orderingName: String?,

    @ColumnInfo(name = PlaylistEntity.PLAYLIST_THUMBNAIL_URL)
    override val thumbnailUrl: String?,

    @ColumnInfo(name = PlaylistEntity.PLAYLIST_DISPLAY_INDEX)
    override var displayIndex: Long?,

    @ColumnInfo(name = PlaylistEntity.PLAYLIST_THUMBNAIL_PERMANENT)
    open val isThumbnailPermanent: Boolean?,

    @ColumnInfo(name = PlaylistEntity.PLAYLIST_THUMBNAIL_STREAM_ID)
    open val thumbnailStreamId: Long?,

    @ColumnInfo(name = PLAYLIST_STREAM_COUNT)
    open val streamCount: Long
) : PlaylistLocalItem {

    override val localItemType: LocalItemType
        get() = LocalItemType.PLAYLIST_LOCAL_ITEM

    companion object {
        const val PLAYLIST_STREAM_COUNT: String = "streamCount"
    }
}
